doc/src/guides/solving/solve-ode.md

```diff
@@ -0,0 +1,423 @@
+(solving-guide-inequalities)=
+# Solve an Ordinary Differential Equation (ODE) Algebraically
+
+Use SymPy to solve an ordinary differential equation (ODE) algebraically. For
+example, solving $y''(x) + 9y(x)=0 $ yields $ y(x)=C_{1} \sin(3x)+ C_{2}
+\cos(3x)$.
+
+## Alternatives to Consider
+- To numerically solve a system of ODEs, use a [SciPy ODE
+  solver](https://docs.scipy.org/doc/scipy/reference/integrate.html#solving-initial-value-problems-for-ode-systems)
+  such as `solve_ivp`. You can also use SymPy to create and then
+  {func}`~.lambdify` an ODE to be solved numerically using SciPy's as
+  `solve_ivp` as described below in [](#numerically-solve-an-ode-in-scipy).
+
+## Solve an Ordinary Differential Equation (ODE)
+
+Here is an example of solving an ordinary differential equation algebraically
+using {func}`~.dsolve`:
+
+```py
+>>> from sympy import Function, dsolve, Derivative
+>>> from sympy.abc import x
+>>> y = Function('y')
+>>> result = dsolve(Derivative(y(x), x, x) + 9*y(x), y(x))
+>>> result
+Eq(y(x), C1*sin(3*x) + C2*cos(3*x))
+```
+
+You can then use SymPy to verify that the solution is correct:
+
+```py
+>>> from sympy import checkodesol
+>>> solution = result.rhs
+>>> solution
+C1*sin(3*x) + C2*cos(3*x)
+>>> checkodesol(Derivative(y(x), x, x) + 9*y(x), solution)
+(True, 0)
+```
+
+The output of {func}`~.checkodesol` is a tuple where the first item, a boolean,
+tells whether substituting the solution into the ODE results in `0`, indicating
+the solution is correct.
+
+## Guidance
+
+### Input Format
+
+There are many ways to express derivatives of functions. For an undefined
+function, both {class}`~.Derivative` and {func}`~.diff` represent the undefined
+derivative. Thus, all of the following `ypp` ("y prime prime") represent $y''$,
+the second derivative with respect to $x$ of a function $y(x)$:
+
+```py
+ypp = y(x).diff(x, x)
+ypp = y(x).diff(x, 2)
+ypp = y(x).diff((x, 2))
+ypp = diff(y(x), x, x)
+ypp = diff(y(x), x, 2)
+ypp = Derivative(y(x), x, x)
+ypp = Derivative(y(x), x, 2)
+ypp = Derivative(Derivative(y(y(x), x), x)
+ypp = diff(diff(y(x), x), x)
+yp = y(x).diff(x)
+ypp = yp.diff(x)
+```
+
+We recommend specifying the function to be solved for, as the second argument to
+{func}`~.dsolve`. Note that it must be a function rather than a variable
+(symbol). SymPy will give an error if you specify a variable ($x$) rather than a
+function ($f(x)$):
+
+```py
+>>> dsolve(Derivative(y(x), x, x) + 9*y(x), x)
+Traceback (most recent call last):
+    ...
+ValueError: dsolve() and classify_ode() only work with functions of one variable, not x
+```
+
+Similarly, you must specify the argument of the function: $f(x)$, not just $f$.
+
+You can define the function to be solved for in two ways. The subsequent syntax
+for specifying initial conditions depends on your choice.
+
+## Define a Function Without Including Its Independent Variable
```

Comment:
I think this is supposed to be level 4 (and the same with the next several headers).

Reply:
My intention was to have

1. Define a Function Without Including Its Independent Variable
2. Define a Function of an Independent Variable

as level 2, with the idea that each is an _option_ (as you suggest below) for an entire approach to using dsolve--setting up `dsolve`, adding initial conditions or boundary conditions, and (for option 2) a caveat. The subsequent sections (Use the Solution Result, etc.) apply to either option 2.

If you think it would be helpful, I could create a parent level 2 and put the two options below it as level 3:
## Options to Define an ODE
### Option 1: Define a Function Without Including Its Independent Variable
### Option 2: Define a Function of an Independent Variable

Reply:
Move to a new h2 Options to Define an ODE
You can define the function to be solved for in two ways. The subsequent syntax for specifying initial conditions depends on your choice.

Could move options 1 and 2 under Guidance instead

Rename "Input Format" to Defining Derivatives